workflow/rules/peak_analysis.smk

```diff
@@ -0,0 +1,209 @@
+        "0.66.0/bio/deeptools/plotfingerprint"
+
+rule macs2_callpeak_broad:
+    input:
+        treatment="results/orphan_rm_sorted/{sample}.bam",
+        control="results/orphan_rm_sorted/{control}.bam"
+    output:
+        # all output-files must share the same basename and only differ by it's extension
+        # Usable extensions (and which tools they implicitly call) are listed here:
+        #         https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/macs2/callpeak.html.
+        multiext("results/macs2_callpeak/{sample}-{control}.broad",
+                 "_peaks.xls",
+                 # these output extensions internally set the --bdg or -B option:
+                 "_treat_pileup.bdg",
+                 "_control_lambda.bdg",
+                 # these output extensions internally set the --broad option:
+                 "_peaks.broadPeak",
+                 "_peaks.gappedPeak"
+                 )
+    log:
+        "logs/macs2/callpeak.{sample}-{control}.broad.log"
+    params: # ToDo: move to config?
+        "--broad --broad-cutoff 0.1 -f BAMPE -g hs --SPMR --qvalue 0.05 --keep-dup all"
+        # ToDo: Update wrapper to check for " --broad$" or " --broad " instead of only "--broad" (line 47),
+        #  then "--broad" in params can be removed here in params
+    wrapper:
+        "0.66.0/bio/macs2/callpeak"
+
+rule macs2_callpeak_narrow:
+    input:
+        treatment="results/orphan_rm_sorted/{sample}.bam",
+        control="results/orphan_rm_sorted/{control}.bam"
+    output:
+        # all output-files must share the same basename and only differ by it's extension
+        # Usable extensions (and which tools they implicitly call) are listed here:
+        #         https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/macs2/callpeak.html.
+        multiext("results/macs2_callpeak/{sample}-{control}.narrow",
+                 "_peaks.xls",
+                 # these output extensions internally set the --bdg or -B option:
+                 "_treat_pileup.bdg",
+                 "_control_lambda.bdg",
+                 # these output extensions internally set the --broad option:
+                 "_peaks.narrowPeak",
+                 "_summits.bed"
+                 )
+    log:
+        "logs/macs2/callpeak.{sample}-{control}.narrow.log"
+    params: # ToDo: move to config?
+        "-f BAMPE -g hs --SPMR --qvalue 0.05 --keep-dup all"
+    wrapper:
+        "0.66.0/bio/macs2/callpeak"
+
+rule peaks_count:
+    input:
+        peaks="results/macs2_callpeak/{sample}-{control}.{peak}_peaks.{peak}Peak"
+    output:
+        "results/macs2_callpeak/peaks_count/{sample}-{control}.{peak}.peaks_count.tsv"
+    log:
+        "logs/macs2_callpeak/peaks_count/{sample}-{control}.{peak}.peaks_count.log"
+    conda:
+        "../envs/gawk.yaml"
+    shell:
+        "cat {input.peaks} | wc -l | gawk -v OFS='\t' '{{print \"{wildcards.sample}-{wildcards.control}_{wildcards.peak}_peaks\", $1}}' > {output} 2> {log}"
```

Comment:
This is somewhat cosmetic, but I would rather improve the readability of this pipe a bit, by stretching it out across some lines:
```suggestion
        "cat {input.peaks} | "
        " wc -l | "
        " gawk -v OFS='\t' '{{print \"{wildcards.sample}-{wildcards.control}_{wildcards.peak}_peaks\", $1}}' "
        " > {output} 2> {log}"
```